### DISCUSSION TITLE
Color shades like `text-primary-500` and `border-error-700`

### DISCUSSION THREAD

**Discussion by mhsnook** [saadeghi/daisyui#1710] [Ideas / feature requests]
DaisyUI does a really good thing for us by converting all the concrete colors like `purple`, `blue`, `green`, `red` into theme colors like `primary`, `info`, `success`, `error`, `secondary`, and `accent`. This is great for theme management, and for switching themes entirely without changing anything but your daisyui section in your tailwind config. Beautiful -- love it -- thank you.

But with these semantic colors we don't get to use the regular tailwind color variations like `text-primary-800` and `bg-primary-100`. So instead I'm using a white background wrapper element and then a `bg-primary/10` (which for some reason _does_ receive the tailwind add-on class treatment 🤔 ), but this is not ideal/means I have to manage two wrappers and put the properties in the right place and just makes that element twice as complex to maintain. 

I've looked through the discussions/issues and seen a few other places where you recommend extending the tw colors (e.g. [here](https://github.com/saadeghi/daisyui/discussions/822)) but if I took this approach it would break the very nice DaisyUI ability to just swap in a new theme without rewriting the colors myself each time. 

p.s. I hope this hasn't been discussed already, I looked around for related issues and didn't see exactly this one, but I think it's potentially a very valuable addition!

**Comment by saadeghi** [comment]
> DaisyUI does a really good thing for us by converting all the concrete colors like `purple`, `blue`, `green`, `red` into theme colors like `primary`, `info`, `success`, `error`, `secondary`, and `accent`

> But with these semantic colors we don't get to use the regular tailwind color variations like `text-primary-800` and `bg-primary-100`.

First I have to correct: It's not converting them. `primary`, `secondary`, etc are **additional** colors (so they can have different values for different themes). Tailwind colors still work (but of course those class names like `bg-purple-500`, `bg-blue-500`, etc are constant colors on all themes)

>if I took this approach it would break the very nice DaisyUI ability to just swap in a new theme without rewriting the colors myself each time.

I don't think it's breaking anything 🤔  
For example, we have a `primary` color (which has color value on all built-in daisyUI themes) and you want to add a new color called `primary-200`, you can easily add it like this 👇  
`tailwind.config.js`:
```js
module.exports = {
  //...
  theme: {
    extend: {
      colors: {
        'primary-200': 'hsl(var(--primary-200) / <alpha-value>)',
      },
    },
  },
}
```
And give it the color value in each theme like this in `tailwind.config.js`: https://play.tailwindcss.com/YitXTO4zGV?file=config  
(the numbers in below example are HSL values, they are same value as primary color of each theme but with increased lightness)
```js
module.exports = {
  //...
  daisyui: {
    themes: [
      {
        light: {
          ...require('daisyui/src/colors/themes')['[data-theme=light]'],
          '--primary-200': '258 94% 81%',
        },
      },
      {
        cupcake: {
          ...require('daisyui/src/colors/themes')['[data-theme=cupcake]'],
          '--primary-200': '183 47% 89%',
        },
      },
      //...
    ],
  },
}
```
Or give it the color values in CSS file like this: https://play.tailwindcss.com/G6uq1RCy3i?file=css
```css
:root,
[data-theme=light] {
  --primary-200: 258 94% 81%;
}
[data-theme=cupcake] {
  --primary-200: 183 47% 89%;
}
```
Every other color still works, you just added a new themeable color name.  

Let me know if you needed any questions.

**Reply by mhsnook** [reply]
Thanks for your response and for this wonderful library!

I appreciate the correction too; in that case I rephrase my feature suggestion to "Add color values for `colorName-Num` for all combinations of semantic color names and number values like tailwind's colors `50, 100, 200, 300, 400, 500, 600, 700, 800, 900`, which also work with utilities like `hover:` and `md:`."

Your suggestion is interesting and I may do it, but it sounds like I will have to learn a lot of new things and write code for each color-value I want to access in my app, and/or maybe also for each theme I want to be able to use them in use(? or maybe it depends on which approach I take). And so far I have been able to entirely avoid that level of complexity and that area of expertise by relying on Tailwind and DaisyUI -- leaving it up to the experts! like you! -- so, I may someday do this, but it really doesn't sort of meet the desire for having these values, like `text-error-800` and `bg-success-100` to be built in from the start. thanks again 🙏

**Reply by saadeghi** [reply]
>but it sounds like I will have to learn a lot of new things and write code for each color-value I want to access in my app

Not really. It's quite the opposite!  
When you have access to all shades of all colors, you have to make design decisions every time you want to use a color and you should consider accessibility of the colors (to make sure the colors you're using for background+foreground have a proper contrast) and then you should consider dark mode and make sure everything you designed also have a good alternative in dark mode.

But with daisyUI colors (disclaimer: diasyUI didn't invent this color system. similar color systems were being used by Material design and many other design systems before) you don't have to worry about contrast, choosing shades, dark mode, etc.  
It's [really simple](https://daisyui.com/docs/colors/)  
1. There is `base-*` color which is the surface color of the page (it's light on light theme and dark on dark theme)
2. You pick 1 to 3 brand colors (`primary`, `secondary`, `accent`). Every time you want to use brand colors, you use one of them
3. There is a `neutral` color, which is a dark color for when you need a dark bold color but you don't want to use a colorful brand color.
4. You can also use 4 state colors (`info`, `success`, `warning`, `error`) if needed

That's it! All these colors can be used as background and whenever you want to use text on them you can use `{COLOR}-content` color on them which gives you a good contrast.  
And the above information is only needed if you want to use color utility classes. You don't need any knowledge for using components like button, etc. because they have the background/foreground colors themselves and you don't need to do anything.

If you try this color names in a project you quickly realize how easier and faster it is, and it doesn't really limit you as you might imagine.

Let me know if you have any questions.

**Reply by mhsnook** [reply]
oh thank you -- I hadn't seen this bit about `{color}-content` and `-focus`. That is really helpful. I think the only thing missing then for me is when I want a really light version of it. This `-100` type thing. [Example here](https://play.tailwindcss.com/wkCoHO4LZr). 

For some situations I want to be able to use this pastel version of a color, especially when I think I am trying to avoid being too "loud" about my use of color, or I want to use color more often but stay in these muted tones to avoid a really high-contrast environment (which can be distracting for people with learning disabilities, cognitive impairments, etc.). I love to use color and I want to use it kind of a lot, but I just want to have this middle ground to choose from. 

I will keep playing with it. HSL seems pretty easy to work with... I just need a way to say "take the primary/accent/error colors and make them all `lightness: 95%`... anyway, that's as far as I've gotten. Thank you for the info and suggestions. Maybe one of them will work for me!

**Comment by mhsnook** [comment]
Hi I have finally reached the point where I have to actually try and solve this for my org so I dug in and you're right the things aren't as complex as I was worried they would be. It's a nice system! I have got it working on this little play link [https://play.tailwindcss.com/ftmiDXNcPy](https://play.tailwindcss.com/ftmiDXNcPy) but it's not working in my actual code (running updated tailwindcss and daisyui). 

Do you have any troubleshooting tips? I can see that the CSS variable is being created, but 
<img width="367" alt="Screenshot 2023-09-21 at 5 52 02 PM" src="https://github.com/saadeghi/daisyui/assets/1102783/4eaea866-af11-468b-bdee-0c0017ecb3f6">

but then the `bg-error-100` class doesn't actually apply the background color... and `bg-error-100` doesn't seem to be emitted as a classname. So I'm a little stumped. Do you have any tips for how to troubleshoot this or step through the build process to find my error? Thanks.

(Also for some reason this broke the use of bg-white in my main css file.)

**Reply by saadeghi** [reply]
Your code is correct.  
If class names (any class name) is working on Tailwind Play but it's not working on your project, it's either a bad [`content` config](https://tailwindcss.com/docs/content-configuration) (`content` is not addressing to all template files) or you're probably using [dynamic class names](https://tailwindcss.com/docs/content-configuration#dynamic-class-names) like (`bg-${{colorName}}`)
